back-end/index.js

```diff
@@ -232,7 +252,7 @@ app.patch('/modules/:moduleId/completed', async (req, res) => {
             });
 
             const courses = await prisma.courses.findMany();
-            recommendationResults = await recommendations(courses, updatedModule.course.difficulty, updatedModule.course.avgRating, updatedModule.course.title );
+            recommendationResults = await recommendations(courses, updatedModule.course.difficulty, updatedModule.course.avgRating, updatedModule.course.title, username );
```

Comment:
OK I see that you updated this to pass in the username. You can ignore the previous comment